rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -350,3 +361,25 @@ def _generate_relations_and_mappings_for_role(self):
                 f"Failed to generate relations and mappings for role {self.role.name}, UUID :{self.role.uuid}: {e}"
             )
             raise DualWriteException(e)
+
+
+def delete_permission(permission: Permission):
```

Comment:
I would probably move this to the seeding handler. It seems like more of a seeding concern. Yes it can involve custom roles, but that is as a result of a system-level "seeded" concept (permission) changing.